src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sources/streamfactories/StreamFactoryBase.scala

```diff
@@ -1,24 +1,18 @@
 package com.microsoft.partnercatalyst.fortis.spark.sources.streamfactories
 
-import com.microsoft.partnercatalyst.fortis.spark.logging.FortisTelemetry
+import com.microsoft.partnercatalyst.fortis.spark.logging.Loggable
 import com.microsoft.partnercatalyst.fortis.spark.sources.streamprovider.{ConnectorConfig, InvalidConnectorConfigException, StreamFactory}
 import org.apache.spark.streaming.StreamingContext
 import org.apache.spark.streaming.dstream.DStream
 
 import scala.reflect.ClassTag
 import scala.util.Try
 
-abstract class StreamFactoryBase[A: ClassTag] extends StreamFactory[A]{
+abstract class StreamFactoryBase[A: ClassTag] extends StreamFactory[A] with Loggable {
```

Comment:
The decorator pattern for logging does not work well for Spark. The problem is that through this approach, the log/telemetry methods become instances methods on the implementing class, meaning that if these methods are used anywhere in functions whose closures are serialized by Spark the entire implementing class will need to be serializable. Today, checkpointing is not enabled, but this change will preclude us from enabling it without making our `StreamFactory`s serializable (which IIRC, is not possible or desirable). 

I understand the desire to have a consistent paradigm for logging and telemetry, but I would recommend rewriting this PR to converge on the old telemetry approach which was chosen originally to address those serialization concerns in a simple way (static / not bound to caller instance).